index.js

```diff
@@ -10,25 +10,23 @@ function generate(argIndex) {
 		var promise = thunk(value);
 		thunk.then = promise.then.bind(promise);
 		thunk.catch = promise.catch.bind(promise);
-		return thunk;
 
-		function thunk(result) {
-			if (promise) {
-				// Prevent unhandled rejection errors if promise is never used, and only used as a thunk
-				promise.catch(function () {});
-			}
+		// Prevent unhandled rejection errors if promise is never used, and only used as a thunk
+		promise.catch(function () {});
```

Comment:
Test code:

```js
delay(5000)
  .then(delay.reject(10))
  .catch(function () {
    console.log('all good!')
  });
```

Old code output (`unhandledrejection` fired first, then the error icon changes to the checkmark after the 5000ms with `rejectionhandled`)

<img width="475" alt="screen shot 2016-05-16 at 16 04 26" src="https://cloud.githubusercontent.com/assets/1402241/15285194/af9c6f14-1b0a-11e6-985a-2b58068d79c1.png">

This PR's output:

<img width="474" alt="screen shot 2016-05-16 at 16 06 45" src="https://cloud.githubusercontent.com/assets/1402241/15285249/dda2f50e-1b0a-11e6-9780-ea7fe3a27307.png">